src/sdk/accountspace_nb.js

```diff
@@ -47,29 +46,28 @@ class AccountSpaceNB {
         account_util._check_if_requesting_account_is_root_account(action, requesting_account,
                 { username: params.username, path: params.iam_path });
         account_util._check_username_already_exists(action, params, requesting_account);
-        const iam_arn = iam_utils.create_arn_for_user(requesting_account._id.toString(), params.username, params.iam_path);
-        const account_name = account_util.get_account_name_from_username(params.username, requesting_account.name.unwrap());
+        const iam_arn = iam_utils.create_arn_for_user(requesting_account._id.toString(), params.username,
+                                    params.iam_path || IAM_DEFAULT_PATH);
+        const account_name = account_util.get_account_name_from_username(params.username, requesting_account._id.toString());
         const req = {
             rpc_params: {
                 name: account_name,
                 email: account_name,
                 has_login: false,
                 s3_access: true,
                 allow_bucket_creation: true,
-                owner: requesting_account._id.toString(),
+                owner: requesting_account._id,
                 is_iam: true,
                 iam_arn: iam_arn,
```

Comment:
You don't need to pass `iam_arn`, right?
It is removed from the schema, also in the function `create_account`.